### ISSUE TITLE
Storage prefixes configurable

### ISSUE THREAD

**Issue by baillyjamy** [aws-amplify/amplify-ios#916]
**Is your feature request related to a problem? Please describe.**
I would like to use amplify-ios lib to retreive S3 object from existing S3 but the prefix "/public" is forced. It is therefore impossible for me to find my files.

**Describe the solution you'd like**
Can you make the prefixes configurable? This would make life easier for existing projects.

**Additional context**
I see that I can add "[pluginOptions](https://github.com/aws-amplify/amplify-ios/blob/main/Amplify/Categories/Storage/Operation/Request/StorageRemoveRequest.swift)" in my request. Maybe I can change/delete the prefix through this? But I can't find the documentation that lists the possible key/values

**Comment by diegocstn** [comment]
hi @baillyjamy ,
thanks for reaching out.
The Amplify Storage plugin currently support three different [access levels](https://docs.amplify.aws/lib/storage/configureaccess/q/platform/js): `public`, `protected`, `private`, defined by the [`StorageAccessLevel` enum](https://github.com/aws-amplify/amplify-ios/blob/main/Amplify/Categories/Storage/StorageAccessLevel.swift).

Each storage request accepts an `options`  argument based on its type.
For example, below a list request:
```
let listOptions = StorageListRequest.Options(accessLevel: .private)
Amplify.Storage.list(options: listOptions) { ... }
```
Unfortunately we don't currently support fully customizable prefixes, but I'll mark this as a possible enhancement and discuss it the team.

**Comment by baillyjamy** [comment]
Thank you for taking the time to read me and consider my request.

I would like to give you some additional feedback. 
I don't think it's a good idea to force people who want to use an aws functionality on mobile to use the Amplify workflow if it's very restrictive, for example in my case, I want to access an S3 instance created without amplify so it doesn't necessarily respect its rules (rules that I don't dispute) but I use Cognito to give access to my S3.
I could use [aws-sdk-ios](https://github.com/aws-amplify/aws-sdk-ios) but it's a bad idea to reinvent the wheel. Amplify facilitates Cognito authentication and renews it. Also, all [aws-sdk-ios](https://github.com/aws-amplify/aws-sdk-ios) documentation refers to Amplify.
So even if I want to re-implement the Cognito authentication system and connect to my S3 with aws-sdk-ios, it's difficult without documentation.

Well, I may be a lazy guy but I tried to bypass Amplify and reinvent the wheel for Android, it was not a good idea. I'd like to avoid starting again here ^^

What I'm trying to say is that either we need an Amplify that is more customisable and less restrictive, or we need an aws-sdk-ios that is better documented and less in the shadows.
Or maybe I didn't understand anything about Amplify? In that case I'd like some help to find out how you would solve my case.

Thanks for reading, I'm not very good English, it's just feedback, not a request. It's not arrogant.

**Comment by richardmcclellan** [comment]
For reference, this has been requested for Android as well: https://github.com/aws-amplify/amplify-android/issues/910

**Comment by baillyjamy** [comment]
Hello, I'm coming back to you to see if you're going to add this feature or not. And if so, do you have an idea of when?
On my side I have to make a choice if I use Amplify or not and I have to move forward on the subject of retrieving my S3 objects. Your answer will help me to make a decision.

**Comment by baillyjamy** [comment]
Hello, being unanswered, I take the liberty of asking you again to find out what you are going to do.

**Comment by saravr** [comment]
These prefixes should be configurable - developers/end users should decide what's best for them. Can this be request be prioritized?

**Comment by danquack** [comment]
The JS client also had this issue and there appears to be a workaround. Wondering if something similar could happen for iOS https://github.com/aws-amplify/amplify-js/issues/332#issuecomment-602606514

**Comment by samulacrum** [comment]
Yeah, this entirely breaks our use case - and we cannot re-configure our existing backend setup just to support the forced prefixes.

Please make this configurable - and make it more explicit in the documentation that this behavior occurs.

**Comment by lawmicha** [comment]
A discussion on [discord](https://discord.com/channels/705853757799399426/707328996995760179/815998384284237844), the developer wants to completely remove the prefix. The flexibliity to configure custom prefixes should allow configuring "empty" or no prefix for certain levels.

**Comment by baillyjamy** [comment]
Still no news for this feature? I don't understand why this feature is implemented for the js lib and not for ios.

**Comment by lawmicha** [comment]
This https://github.com/aws-amplify/amplify-ios/pull/1277 been released in https://github.com/aws-amplify/amplify-ios/releases/tag/v1.14.0 Please let us know if you have any questions
